packages/langium/src/references/linker.ts

```diff
@@ -107,13 +108,16 @@ export class DefaultLinker implements Linker {
             try {
                 const refId = getReferenceId(info.container.$type, info.property);
                 const description = this.getCandidate(info.container, refId, ref);
-                if (!isLinkingError(description) && this.langiumDocuments().hasDocument(description.documentUri)) {
-                    // The target document is already loaded
-                    const linkedNode = this.loadAstNode(description);
-                    ref._ref = linkedNode ?? this.createLinkingError(info, refId, description);
-                } else {
-                    // The target document is not loaded yet, or the target was not found in the scope
+                if (isLinkingError(description)) {
                     ref._ref = description;
+                } else {
+                    ref._nodeDescription = description;
+                    if (this.langiumDocuments().hasDocument(description.documentUri)) {
+                        // The target document is already loaded
+                        const linkedNode = this.loadAstNode(description);
+                        ref._ref = linkedNode ?? this.createLinkingError(info, refId, description);
+                    }
+                    // The target document is not loaded yet
```

Comment:
This comment doesn't make much sense here (it would if we'd have an `else` block).